src/platform/i386/boot_comp.c

```diff
@@ -15,14 +15,17 @@ extern u8_t *boot_comp_pgd;
 int boot_nptes(unsigned int sz) { return round_up_to_pow2(sz, PGD_RANGE)/PGD_RANGE; }
 
 int
-boot_pgtbl_mappings_add(struct captbl *ct, pgtbl_t pgtbl, capid_t ptecap, const char *label,
+boot_pgtbl_mappings_add(struct captbl *ct, capid_t pgdcap, capid_t ptecap, const char *label,
```

Comment:
the type "pgtbl_t" is defined as `typedef struct pgtbl * pgtbl_t;`
More specific, we can see the code `pgtbl_t pgtbl = (pgtbl_t)chal_va2pa(&boot_comp_pgd);`
So "pgtbl_t" is the physical address of the page table.
I change the argument to use capability instead of physical address.
We extensively use "capid_t" everywhere in the kernel to refer to a capability.
Why do you say it is ambiguous?